pkg/collector/corechecks/sbom/check.go

```diff
@@ -140,14 +142,18 @@ func (c *Check) Run() error {
 		),
 	)
 
+	// Trigger an initial scan on host
+	c.processor.processHostRefresh()
+
 	imgRefreshTicker := time.NewTicker(time.Duration(c.instance.PeriodicRefreshSeconds) * time.Second)
 
 	for {
 		select {
 		case eventBundle := <-imgEventsCh:
-			c.processor.processEvents(eventBundle)
+			c.processor.processContainerImagesEvents(eventBundle)
 		case <-imgRefreshTicker.C:
-			c.processor.processRefresh(c.workloadmetaStore.ListImages())
+			c.processor.processContainerImagesRefresh(c.workloadmetaStore.ListImages())
+			c.processor.processHostRefresh()
```

Comment:
The "problem" with the Scanner approach is that the set of analyzers may be different between host / container scans (it may also be different between 2 container images).
So I suggest we go for the dedicated Goroutine with its own scanner for now. WDYT ?

Reply:
I introduced a new Scanner, that have both host and image scan request channels. The requester passes a callback that gets called when SBOM was generated.
Regarding the refresh period, I'm not even sure why we periodically rescan the image. Is it to maintain a kind of heartbeat for SBOM ?